broker-core/src/main/java/org/apache/qpid/server/security/UserConnectionLimitProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.qpid.server.security;
+
+import org.apache.qpid.server.logging.EventLoggerProvider;
+
+@FunctionalInterface
+public interface UserConnectionLimitProvider
+{
+    UserConnectionLimiter getLimiter(EventLoggerProvider eventLoggerProvider);
```

Comment:
This leads to the sharing of the connection counters among VistualHosts, because broker connection limit providers are common for all  VistualHosts. But, it is stated in documentation that VistualHosts are independent, hence every VistualHosts needs it's own ConnectionLimiter with independent connection counters. ConnectionLimitProvider works as factory that build a new independent limiter for each VirtualHost.